jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -298,6 +299,73 @@ public EndPoint getEndPoint()
         return _endPoint;
     }
 
+    /**
+     * <p>Return the Local Name of the connected channel.</p>
+     *
+     * <p>
+     * This is the name after the connector is bound and the connection is accepted.
+     * The value is typically the local address and optionally the local host name
+     * if the local address is unavailable.
+     * </p>
+     * <p>
+     * Value can be overridden by {@link AbstractConnector#setLocalAuthority(HostPort)} for
+     * scenarios where Jetty is being an intermediary and the local name
+     * needs to be changed to satisfy public (pre-intermediary) HTTP behaviors
+     * such as absolute-URI creation (eg: Location response header).
+     * </p>
+     *
+     * @return the local name if overridden, or the local address, or
+     * null in the case of no local address (usually seen in connectors not based on IP networking).
+     */
+    public String getLocalName()
+    {
+        Connector connector = getConnector();
+
+        if (connector instanceof AbstractConnector)
+        {
+            HostPort localAuthority = ((AbstractConnector)connector).getLocalAuthority();
+            if (localAuthority != null)
+                return localAuthority.getHost();
+        }
+
+        InetSocketAddress local = getLocalAddress();
+        if (local != null)
+            return local.getHostString();
+
+        return null;
+    }
+
+    /**
+     * <p>Return the Local Port of the connected channel.</p>
+     *
+     * <p>
+     * This is the port the connector is bound to and is accepting connection on.
+     * </p>
+     * <p>
+     * Value can be overridden by {@link AbstractConnector#setLocalAuthority(HostPort)} for
+     * scenarios where Jetty is being an intermediary and the local port
+     * needs to be changed to satisfy public (pre-intermediary) HTTP behaviors
+     * such as absolute-URI creation (eg: Location response header).
+     * </p>
+     *
+     * @return the local authority port if overridden, or the local address port, or
+     * 0 in the case of no local address (usually seen in connectors not based on IP networking).
+     */
+    public int getLocalPort()
+    {
+        Connector connector = getConnector();
+
+        if (connector instanceof AbstractConnector)
```

Comment:
Ditto, test should be for `NetworkConnector` which should always provide the local port

Reply:
But also let's not convert to Host/Port here.  Might need to leave these as deprecated, but pass the local Authority all the way to the `Request`